### PR TITLE
Hog/log metrics

### PR DESCRIPTION
**Pull request by ericlee0803**
Added:
- compute_metrics.py
- check learning task
- log training metrics
- log_validation metrics

### CODE REVIEW

File: sigopt/xgboost/run.py [hunk 1]
```diff
@@ -154,7 +167,21 @@ def train_xgb(self):
       if self.run_options_parsed['log_stderr']:
         log_dict["stderr"] = stderr
       self.run.set_logs(log_dict)
-    return bst
+    self.bst = bst
+
+  def log_training_metrics(self):
```
[the remark this answers is not in the document]
Comment (ericlee0803) [reply sigopt/xgboost/run.py:178]: Done, see: https://app.sigopt.com/run/114071

**Review by harvey-intel — APPROVED** [review]
looks good to me. Can you add metric logging to the [integration test](https://github.com/sigopt/sigopt-python/blob/hog/main/integration_test/test_xgboost.py)? Might need to rebase on `hog/main` to pick up the new changes. Otherwise, it's good to merge.

**Comment by ericlee0803** [comment]
going to merge in real quick and then add the integration test in a bit, that way yuhong can rebase his pr

File: setup.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
 with open(os.path.join(here, 'requirements-dev.txt')) as requirements_dev_fp:
   dev_install_requires = requirements_dev_fp.read().split('\n')
 
-xgboost_install_requires = ['xgboost>=1.3.1', 'scikit-learn']
+xgboost_install_requires = ['xgboost>=1.3.1', 'scikit-learn', 'numpy']
```
Comment (dwanderson-intel) [inline setup.py:31]: fwiw we should at least have minimum versions for this. you don't need to go and find the absolute bare minimum; just use whatever's getting installed and I think that's fine

**Pull request merged**